### ISSUE TITLE
Zoom pdf programmatically?

### ISSUE THREAD

**Issue by rjtubera** [react-pdf-viewer/react-pdf-viewer#406]
Hi is there any way to zoom the pdf programmatically? I can't find it in the documentation

Thanks

**Comment by rjtubera** [comment]
I need to change the zoom when they click the fullscreen button. Thanks

**Comment by phuoc-ng** [comment]
There are 2 ways to handle that. I'll get back to you in next few hours after preparing all solutions.

**Comment by rjtubera** [comment]
@phuoc-ng, Thanks man!

**Comment by phuoc-ng** [comment]
@rjtubera 
There are 2 approach to archive your requirement.

## 1. Create our own plugin

You can use this approach with the current version 2.3.0.
For the full example code, please take a look at this [example](https://github.com/react-pdf-viewer/examples/tree/main/zoom-after-full-screen).

_Usage_ ([link](https://github.com/react-pdf-viewer/examples/blob/main/zoom-after-full-screen/ZoomAfterFullScreenExample.tsx))

```js
import { Viewer } from '@react-pdf-viewer/core';
import { defaultLayoutPlugin } from '@react-pdf-viewer/default-layout';

import '@react-pdf-viewer/core/lib/styles/index.css';
import '@react-pdf-viewer/default-layout/lib/styles/index.css';

import zoomAfterFullScreen from './zoomAfterFullScreen';

interface ZoomAfterFullScreenExampleProps {
    fileUrl: string;
}

const ZoomAfterFullScreenExampleProps: React.FC<ZoomAfterFullScreenExampleProps> = ({ fileUrl }) => {
    const defaultLayoutPluginInstance = defaultLayoutPlugin();
    const zoomAfterFullScreenInstance = zoomAfterFullScreen();

    return (
        <div
            style={{
                height: '100%',
            }}
        >
            <Viewer
                fileUrl={fileUrl}
                plugins={[
                    defaultLayoutPluginInstance,
                    zoomAfterFullScreenInstance,
                ]}
            />
        </div>
    );
};
```

## 2. Wait for the 2.3.1

I see this requirement is quite interesting and could be useful for other people, so I improved (PR #414).
The enhancement will be included in the next 2.3.1.

_Use the [full-screen plugin](https://react-pdf-viewer.dev/plugins/full-screen)_

```js
import { SpecialZoomLevel } from '@react-pdf-viewer/core';
import { fullScreenPlugin } from '@react-pdf-viewer/full-screen';

const fullScreenPluginInstance = fullScreenPlugin({
    // Zoom to fit the screen after entering and exiting the full screen mode
    onEnterFullScreen: (zoom) => {
        zoom(SpecialZoomLevel.PageFit);
    },
    onExitFullScreen: (zoom) => {
        zoom(SpecialZoomLevel.PageFit);
    },
});
```

_Use the [default-layout plugin](https://react-pdf-viewer.dev/plugins/default-layout)_

```js
mport { SpecialZoomLevel, Viewer, Worker } from '@react-pdf-viewer/core';
import { defaultLayoutPlugin } from '@react-pdf-viewer/default-layout';
import { Zoom } from '@react-pdf-viewer/full-screen';

import '@react-pdf-viewer/core/lib/styles/index.css';
import '@react-pdf-viewer/default-layout/lib/styles/index.css';

const App = () => {
    const defaultLayoutPluginInstance = defaultLayoutPlugin({
        toolbarPlugin: {
            fullScreenPlugin: {
                onEnterFullScreen: (zoom: Zoom) => {
                    zoom(SpecialZoomLevel.PageFit);
                },
                onExitFullScreen: (zoom: Zoom) => {
                    zoom(SpecialZoomLevel.PageFit);
                },
            }
        }
    });

    return (
        <Viewer
            fileUrl="..."
            plugins={[
                defaultLayoutPluginInstance,
            ]}
        />
    );
};
```

**Comment by phuoc-ng** [comment]
@rjtubera 
v2.3.1 is released. See [Zoom to fit the screen after entering the full screen mode](https://react-pdf-viewer.dev/examples/zoom-to-fit-the-screen-after-entering-the-full-screen-mode) example.
